### PR TITLE
feat(DynamicPage): initial component implementation

### PR DESCRIPTION
**Pull request by kleinju1017**
**Thank you for your contribution!** 👏

To get it merged faster, kindly review the checklist below:

## Pull Request Checklist
- [x] Reviewed the [Contributing Guidelines](https://github.com/SAP/ui5-webcomponents-react/blob/master/CONTRIBUTING.md)
    + Especially the [How to Contribute](https://github.com/SAP/ui5-webcomponents-react/blob/master/CONTRIBUTING.md#contribute-code) section 
- [x] [Correct commit message style](https://github.com/SAP/ui5-webcomponents-react/blob/master/docs/Guidelines.md#commit-message-style)

### CODE REVIEW

File: packages/main/src/components/DynamicPageAnchorBar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    headerPinned,
+    setHeaderPinned,
+    onToggleHeaderContentVisibility,
+    onHoverToggleButton,
+    style
+  } = props;
+
+  const classes = useStyles();
+
+  const shouldRenderHideHeaderButton = showHideHeaderButton;
```
Comment (Lukas742) [inline packages/main/src/components/DynamicPageAnchorBar/index.tsx:94]: unnecessary reassignment

File: packages/main/src/components/DynamicPage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    anchorBarRef,
+    { noHeader: false }
+  );
+
+  if (headerState === HEADER_STATES.HIDDEN || headerState === HEADER_STATES.HIDDEN_PINNED) {
+    dynamicPageClasses.put(classes.headerCollapsed);
+  }
+
+  const onToggleHeaderContentVisibility = useCallback(
+    (e, element?) => {
```
Comment (Lukas742) [inline packages/main/src/components/DynamicPage/index.tsx:115]:
```suggestion
    (e, element?: HTMLElement) => {
```

File: packages/main/src/components/DynamicPage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+      dynamicPageRef.current.addEventListener(
+        'scroll',
+        () => {
+          if (headerStateRef.current !== HEADER_STATES.VISIBLE_PINNED && headerStateResetOnScroll)
+            setHeaderStateRef(HEADER_STATES.AUTO);
+        },
+        { once: true }
+      );
+    },
+    [dynamicPageRef, classes.headerCollapsed, headerContentHeight, topHeaderHeight]
```
Comment (Lukas742) [inline packages/main/src/components/DynamicPage/index.tsx:138]:
```suggestion
    [dynamicPageRef.current, headerStateRef.current, classes.headerCollapsed]
```

File: packages/main/src/components/DynamicPageAnchorBar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      backgroundImage: `linear-gradient(to left, ${ThemingParameters.sapHighlightColor}, rgba(8,84,160,0))`
+    },
+    '&:after': {
+      backgroundImage: `linear-gradient(to right, ${ThemingParameters.sapHighlightColor}, rgba(8,84,160,0))`,
+      left: '100%'
+    }
+  },
+  anchorBarActionButtonExpandable: {},
+  anchorBarActionButtonPinnable: {},
+  anchorBarActionPinnableAndExandable: {
```
Comment (Lukas742) [inline packages/main/src/components/DynamicPageAnchorBar/index.tsx:54]: typo
```suggestion
  anchorBarActionPinnableAndExpandable: {
```

File: packages/main/src/components/DynamicPageAnchorBar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  );
+
+  return (
+    <section style={style} role="navigation" className={classes.anchorBarActionButton} ref={ref}>
+      {shouldRenderHideHeaderButton && (
+        <Button
+          icon={headerContentHeight === 0 ? 'slim-arrow-down' : 'slim-arrow-up'}
+          data-ui5wcr-dynamic-page-header-action=""
+          className={`${classes.anchorBarActionButton} ${classes.anchorBarActionButtonExpandable} ${
+            showBothActions ? classes.anchorBarActionPinnableAndExandable : ''
```
Comment (Lukas742) [inline packages/main/src/components/DynamicPageAnchorBar/index.tsx:112]: typo
```suggestion
            showBothActions ? classes.anchorBarActionPinnableAndExpandable : ''
```

File: packages/main/src/components/DynamicPageAnchorBar/index.tsx [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+          onMouseOver={onHoverToggleButton}
+          onMouseLeave={onHoverToggleButton}
+        />
+      )}
+      {shouldRenderHeaderPinnableButton && (
+        <ToggleButton
+          icon="pushpin-off"
+          data-ui5wcr-dynamic-page-header-action=""
+          className={`${classes.anchorBarActionButton} ${classes.anchorBarActionButtonPinnable} ${
+            showBothActions ? classes.anchorBarActionPinnableAndExandable : ''
```
Comment (Lukas742) [inline packages/main/src/components/DynamicPageAnchorBar/index.tsx:124]: typo
```suggestion
            showBothActions ? classes.anchorBarActionPinnableAndExpandable : ''
```

File: packages/main/src/components/DynamicPageTitle/index.tsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+  return (
+    <FlexBox
+      className={containerClasses.className}
+      style={style}
+      ref={ref}
+      tooltip={tooltip}
+      data-component-name="DynamicPageTitle"
+      {...passThroughProps}
+      onClick={onToggleHeaderContentVisibility}
```
Comment (Lukas742) [inline packages/main/src/components/DynamicPageTitle/index.tsx:89]: What effect does this have?

**Review by Lukas742 — DISMISSED** [review]
1. The `footer` and `showFooter` prop are not implemented. Only the type definitions are set.
2. When `alwaysShowContentHeader` is true, following error will be displayed: 
```
StoryRenderer.js:424 Warning: `NaN` is an invalid value for the `top` css style property.
```
3. Could you add some more examples to the demo? E.g.:
    - With custom footer/header
    - With custom title (`VariantManagement`)
    - With customizable `DynamicPageTitle`, `DynamicPageHeader` (see `List` component for reference)
4. `DynamicPageAnchorBar` is only used internally. Wouldn't it be better to move the component to the `DynamicPage` folder? What do you think?

Optional:
Could you add some more tests? E.g.: 
- expand/collapse behavior
- showFooter/alwaysShowContentHeader/showHideHeaderButton

**Review by MarcusNotheis — APPROVED** [review]

**Pull request merged**